src/v/coproc/event_listener.cc

```diff
@@ -147,6 +70,17 @@ ss::future<> event_listener::start() {
     return ss::now();
 }
 
+void event_listener::register_handler(
+  coproc_type event_type, event_handler* handler) {
+    _handlers[event_type] = handler;
+    vlog(
+      coproclog.info,
+      "Register new handler for wasm_event_type: {}",
+      coproc_type_as_string_view(event_type));
+}
+
+ss::abort_source& event_listener::get_abort_source() { return _abort_source; }
```

Comment:
maybe _abort_source should be a ss::lw_share_ptr<ss::abort_source> if you want to share it this way?

Reply:
done